---
layout: post
title: "RNAseq教程(1.4)"
subtitle: "RNAseq tutorial(1.4) "
author: "zhouxiaozhao"
catalog: true
tags:
   - bioinformatics
   - RNAseq
---

## 目录

**1.Module 1 - Introduction to RNA sequencing**

1. [Installation](https://www.zhouxiaozhao.cn/2020/11/17/RNAseq(1)/)
2. [Reference Genomes](https://www.zhouxiaozhao.cn/2020/11/28/RNAseq(2)/)
3. [Annotations](https://www.zhouxiaozhao.cn/2020/12/01/RNAseq(3)/)
4. [Indexing](https://www.zhouxiaozhao.cn/2020/12/03/RNAseq(4)/)
5. [RNA-seq Data](https://www.zhouxiaozhao.cn/2020/12/05/RNAseq(5)/)
6. [Pre-Alignment QC](https://www.zhouxiaozhao.cn/2020/12/08/RNAseq(6)/)

**2.Module 2 - RNA-seq Alignment and Visualization**

1. [Adapter Trim](https://www.zhouxiaozhao.cn/2020/12/10/RNAseq(7)/)
2. [Alignment](https://www.zhouxiaozhao.cn/2020/12/12/RNAseq(8)/)
3. [IGV](https://www.zhouxiaozhao.cn/2020/12/15/RNAseq(9)/)
4. [Alignment Visualization](https://www.zhouxiaozhao.cn/2020/12/17/RNAseq(10)/)
5. [Alignment QC](https://www.zhouxiaozhao.cn/2020/12/19/RNAseq(11)/)

**3.Module 3 - Expression and Differential Expression**

1. [Expression](https://www.zhouxiaozhao.cn/2020/12/22/RNAseq(12)/)
2. [Differential Expression](https://www.zhouxiaozhao.cn/2020/12/24/RNAseq(13)/)
3. [DE Visualization](https://www.zhouxiaozhao.cn/2020/12/26/RNAseq(14)/)
4. [Kallisto for Reference-Free Abundance Estimation](https://www.zhouxiaozhao.cn/2020/12/29/RNAseq(15)/)

**4.Module 4 - Isoform Discovery and Alternative Expression**

1. [Reference Guided Transcript Assembly](https://www.zhouxiaozhao.cn/2020/12/31/RNAseq(16)/)
2. [de novo Transcript Assembly](https://www.zhouxiaozhao.cn/2021/01/02/RNAseq(17)/)
3. [Transcript Assembly Merge](https://www.zhouxiaozhao.cn/2021/01/05/RNAseq(18)/)
4. [Differential Splicing](https://www.zhouxiaozhao.cn/2021/01/07/RNAseq(19)/)
5. [Splicing Visualization](https://www.zhouxiaozhao.cn/2021/01/09/RNAseq(20)/)

**5.Module 5 - De novo transcript reconstruction**

1. [De novo RNA-Seq Assembly and Analysis Using Trinity](https://www.zhouxiaozhao.cn/2021/01/12/RNAseq(21)/)

**6.Module 6 - Functional Annotation of Transcripts**

1. [Functional Annotation of Assembled Transcripts Using Trinotate](https://www.zhouxiaozhao.cn/2021/01/14/RNAseq(22)/)

## 1.4 Indexing

### 创建HISAT索引

为chr22和ERCC spikein序列创建HISAT2索引。HISAT2可以将外显子和剪接位点合并到索引文件中进行对齐。首先创建一个剪接站点文件，然后创建一个外显子文件。最后制作FM索引。

要了解更多关于HISAT2索引策略与其他软件的不同之处，请参阅[说明书](https://pubmed.ncbi.nlm.nih.gov/25751142/)。

```
hisat2_extract_splice_sites.py chr22_with_ERCC92.gtf >INDEX/splicesites.tsv
hisat2_extract_exons.py chr22_with_ERCC92.gtf >INDEX/exons.tsv
hisat2-build -p 8 --ss INDEX/splicesites.tsv --exon INDEX/exons.tsv chr22_with_ERCC92.fa INDEX/
```

[可选]为所有染色体创建索引，而不是仅仅为chr22，你会做以下操作:

注意:下面的例子没有利用将剪接位点和外显子添加到索引。如果需要，可以使用完整的GTF生成这些文件，并使用适当的选项将它们添加到命令中。

警告:为了索引整个人类基因组，HISAT2需要160GB内存。AWS实例大小将耗尽RAM。

```
#cd /home/ubuntu/workspace/data/fasta/GRCh38
#hisat2-build -p 8 Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa Homo_sapiens.GRCh38.dna_sm.primary_assembly
```