---
layout: post
title: "RNAseq教程(1.1)"
subtitle: "RNAseq tutorial(1.1) "
author: "zhouxiaozhao"
catalog: true
tags:
   - bioinformatics
   - RNAseq
---



## 前言

本教程来自与我保存在[github](https://github.com/Bioinformatics-rookie/rnaseq_tutorial)上的RNAseq教程

这是一个RNA-seq分析的教学教程和工作演示流程，包括介绍云计算(不介绍了，直接从第二章开始)、下一代序列文件格式、参考基因组、基因注释、表达分析、差异表达分析、选择性剪接分析、数据可视化和解释。

## 目录

**1.Module 1 - Introduction to RNA sequencing**

1. [Installation](https://www.zhouxiaozhao.cn/2020/11/17/RNAseq(1)/)
2. [Reference Genomes](https://www.zhouxiaozhao.cn/2020/11/28/RNAseq(2)/)
3. [Annotations](https://www.zhouxiaozhao.cn/2020/12/01/RNAseq(3)/)
4. [Indexing](https://www.zhouxiaozhao.cn/2020/12/03/RNAseq(4)/)
5. [RNA-seq Data](https://www.zhouxiaozhao.cn/2020/12/05/RNAseq(5)/)
6. [Pre-Alignment QC](https://www.zhouxiaozhao.cn/2020/12/08/RNAseq(6)/)

**2.Module 2 - RNA-seq Alignment and Visualization**

1. [Adapter Trim](https://www.zhouxiaozhao.cn/2020/12/10/RNAseq(7)/)
2. [Alignment](https://www.zhouxiaozhao.cn/2020/12/12/RNAseq(8)/)
3. [IGV](https://www.zhouxiaozhao.cn/2020/12/15/RNAseq(9)/)
4. [Alignment Visualization](https://www.zhouxiaozhao.cn/2020/12/17/RNAseq(10)/)
5. [Alignment QC](https://www.zhouxiaozhao.cn/2020/12/19/RNAseq(11)/)

**3.Module 3 - Expression and Differential Expression**

1. [Expression](https://www.zhouxiaozhao.cn/2020/12/22/RNAseq(12)/)
2. [Differential Expression](https://www.zhouxiaozhao.cn/2020/12/24/RNAseq(13)/)
3. [DE Visualization](https://www.zhouxiaozhao.cn/2020/12/26/RNAseq(14)/)
4. [Kallisto for Reference-Free Abundance Estimation](https://www.zhouxiaozhao.cn/2020/12/29/RNAseq(15)/)

**4.Module 4 - Isoform Discovery and Alternative Expression**

1. [Reference Guided Transcript Assembly](https://www.zhouxiaozhao.cn/2020/12/31/RNAseq(16)/)
2. [de novo Transcript Assembly](https://www.zhouxiaozhao.cn/2021/01/02/RNAseq(17)/)
3. [Transcript Assembly Merge](https://www.zhouxiaozhao.cn/2021/01/05/RNAseq(18)/)
4. [Differential Splicing](https://www.zhouxiaozhao.cn/2021/01/07/RNAseq(19)/)
5. [Splicing Visualization](https://www.zhouxiaozhao.cn/2021/01/09/RNAseq(20)/)

**5.Module 5 - De novo transcript reconstruction**

1. [De novo RNA-Seq Assembly and Analysis Using Trinity](https://www.zhouxiaozhao.cn/2021/01/12/RNAseq(21)/)

**6.Module 6 - Functional Annotation of Transcripts**

1. [Functional Annotation of Assembled Transcripts Using Trinotate](https://www.zhouxiaozhao.cn/2021/01/14/RNAseq(22)/)

## 软件安装（module 1： Installation）

分析所需的软件有:samtools, bamo -readcount, HISAT2, stringtie, gffcompare, htseq-count, flexbar, R, ballgown，fastqc和picard-tools。

设置软件安装位置:

```
mkdir student_tools
cd student_tools
```

#### SAMtools

```
wget https://github.com/samtools/samtools/releases/download/1.9/samtools-1.9.tar.bz2
bunzip2 samtools-1.9.tar.bz2
tar -xvf samtools-1.9.tar
cd samtools-1.9
make
./samtools
```

#### bam-readcount

```
export SAMTOOLS_ROOT=$RNA_HOME/student_tools/samtools-1.9
git clone https://github.com/genome/bam-readcount.git
cd bam-readcount
cmake -Wno-dev $RNA_HOME/student_tools/bam-readcount
make
./bin/bam-readcount
```

#### HISAT2

```
wget ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/downloads/hisat2-2.1.0-Linux_x86_64.zip
unzip hisat2-2.1.0-Linux_x86_64.zip
cd hisat2-2.1.0
./hisat2
```

#### StringTie

```
wget http://ccb.jhu.edu/software/stringtie/dl/stringtie-1.3.4d.Linux_x86_64.tar.gz
tar -xzvf stringtie-1.3.4d.Linux_x86_64.tar.gz
cd stringtie-1.3.4d.Linux_x86_64
./stringtie
```

#### gffcompare

```
wget http://ccb.jhu.edu/software/stringtie/dl/gffcompare-0.10.6.Linux_x86_64.tar.gz
tar -xzvf gffcompare-0.10.6.Linux_x86_64.tar.gz
cd gffcompare-0.10.6.Linux_x86_64
./gffcompare
```

#### htseq-count

```
wget https://github.com/simon-anders/htseq/archive/release_0.11.0.tar.gz
tar -zxvf release_0.11.0.tar.gz
cd htseq-release_0.11.0/
python setup.py install --user
chmod +x scripts/htseq-count
./scripts/htseq-count
```

#### TopHat

```
wget https://ccb.jhu.edu/software/tophat/downloads/tophat-2.1.1.Linux_x86_64.tar.gz
tar -zxvf tophat-2.1.1.Linux_x86_64.tar.gz
cd tophat-2.1.1.Linux_x86_64/
./gtf_to_fasta
```

#### kallisto

```
wget https://github.com/pachterlab/kallisto/releases/download/v0.44.0/kallisto_linux-v0.44.0.tar.gz
tar -zxvf kallisto_linux-v0.44.0.tar.gz
cd kallisto_linux-v0.44.0/
./kallisto
```

#### FastQC

```
wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.8.zip --no-check-certificate
unzip fastqc_v0.11.8.zip
cd FastQC/
chmod 755 fastqc
./fastqc --help
```

#### MultiQC

```
pip3 install multiqc
multiqc --help
```

#### Picard

```
wget https://github.com/broadinstitute/picard/releases/download/2.18.15/picard.jar -O picard.jar
java -jar $RNA_HOME/student_tools/picard.jar
```

#### Flexbar

```
wget https://github.com/seqan/flexbar/releases/download/v3.4.0/flexbar-3.4.0-linux.tar.gz
tar -xzvf flexbar-3.4.0-linux.tar.gz
cd flexbar-3.4.0-linux/
export LD_LIBRARY_PATH=$RNA_HOME/student_tools/flexbar-3.4.0-linux:$LD_LIBRARY_PATH
./flexbar
```

#### Regtools

```
git clone https://github.com/griffithlab/regtools
cd regtools/
mkdir build
cd build/
cmake ..
make
./regtools
```

#### RSeQC

```
pip install RSeQC
read_GC.py
```

#### R Libraries

```
#install.packages(c("devtools","dplyr","gplots","ggplot2"),repos="http://cran.us.r-project.org")
#quit(save="no")
```

#### Bioconductor

```
#source("http://bioconductor.org/biocLite.R")
#biocLite(c("genefilter","ballgown","edgeR","GenomicRanges","rhdf5","biomaRt"))
#quit(save="no")
```

#### Sleuth

```
#install.packages("devtools")
#devtools::install_github("pachterlab/sleuth")
#quit(save="no")
```

## 练习

在student_tools下安装bedtools，并编译和测试
