---
layout: post
title: "RNAseq教程(4.3)"
subtitle: "RNAseq tutorial(4.3) "
author: "zhouxiaozhao"
catalog: true
tags:
   - bioinformatics
   - RNAseq
---

## 目录

**1.Module 1 - Introduction to RNA sequencing**

1. [Installation](https://www.zhouxiaozhao.cn/2020/11/17/RNAseq(1)/)
2. [Reference Genomes](https://www.zhouxiaozhao.cn/2020/11/28/RNAseq(2)/)
3. [Annotations](https://www.zhouxiaozhao.cn/2020/12/01/RNAseq(3)/)
4. [Indexing](https://www.zhouxiaozhao.cn/2020/12/03/RNAseq(4)/)
5. [RNA-seq Data](https://www.zhouxiaozhao.cn/2020/12/05/RNAseq(5)/)
6. [Pre-Alignment QC](https://www.zhouxiaozhao.cn/2020/12/08/RNAseq(6)/)

**2.Module 2 - RNA-seq Alignment and Visualization**

1. [Adapter Trim](https://www.zhouxiaozhao.cn/2020/12/10/RNAseq(7)/)
2. [Alignment](https://www.zhouxiaozhao.cn/2020/12/12/RNAseq(8)/)
3. [IGV](https://www.zhouxiaozhao.cn/2020/12/15/RNAseq(9)/)
4. [Alignment Visualization](https://www.zhouxiaozhao.cn/2020/12/17/RNAseq(10)/)
5. [Alignment QC](https://www.zhouxiaozhao.cn/2020/12/19/RNAseq(11)/)

**3.Module 3 - Expression and Differential Expression**

1. [Expression](https://www.zhouxiaozhao.cn/2020/12/22/RNAseq(12)/)
2. [Differential Expression](https://www.zhouxiaozhao.cn/2020/12/24/RNAseq(13)/)
3. [DE Visualization](https://www.zhouxiaozhao.cn/2020/12/26/RNAseq(14)/)
4. [Kallisto for Reference-Free Abundance Estimation](https://www.zhouxiaozhao.cn/2020/12/29/RNAseq(15)/)

**4.Module 4 - Isoform Discovery and Alternative Expression**

1. [Reference Guided Transcript Assembly](https://www.zhouxiaozhao.cn/2020/12/31/RNAseq(16)/)
2. [de novo Transcript Assembly](https://www.zhouxiaozhao.cn/2021/01/02/RNAseq(17)/)
3. [Transcript Assembly Merge](https://www.zhouxiaozhao.cn/2021/01/05/RNAseq(18)/)
4. [Differential Splicing](https://www.zhouxiaozhao.cn/2021/01/07/RNAseq(19)/)
5. [Splicing Visualization](https://www.zhouxiaozhao.cn/2021/01/09/RNAseq(20)/)

**5.Module 5 - De novo transcript reconstruction**

1. [De novo RNA-Seq Assembly and Analysis Using Trinity](https://www.zhouxiaozhao.cn/2021/01/12/RNAseq(21)/)

**6.Module 6 - Functional Annotation of Transcripts**

1. [Functional Annotation of Assembled Transcripts Using Trinotate](https://www.zhouxiaozhao.cn/2021/01/14/RNAseq(22)/)

## 4.3 Stringle Merge

使用Stringtie将所有库中的预测转录本合并成一个统一的转录组。参考Stringtie手册获得更详细的解释：https://ccb.jhu.edu/software/stringtie/index.shtml?t=manual

Options specified below:

- "assembly_GTF_list.txt" is a text file "manifest" with a list (one  per line) of GTF files that you would like to merge together into a  single GTF file.

- '-p 8' tells stringtie to use eight CPUs

- '-o' tells stringtie to write output to a particular file or directory

- '-G' tells stringtie where to find reference gene annotations. It  will use these annotations to gracefully merge novel isoforms (for de  novo runs) and known isoforms and maximize overall assembly quality.

合并所有6个Stringtie结果：

```
ls -1 *Rep*/transcripts.gtf > assembly_GTF_list.txt
cat assembly_GTF_list.txt
stringtie --merge -p 8 -o stringtie_merged.gtf -G $RNA_REF_GTF assembly_GTF_list.txt
```

```
stringtie --merge -p 8 -o stringtie_merged.gtf -G ../chr22_with_ERCC92.gtf assembly_GTF_list.txt
```

结果的转录本是什么样子的?

```
awk '{if($3=="transcript") print}' stringtie_merged.gtf | cut -f 1,4,9 | less
```

将参考引导的转录本与已知的注释进行比较。这使我们能够评估组装RNA-seq数据的转录预测的质量。更多细节，请参考Stringtie GFF和Cuffcompare手册。

```
gffcompare -r ../chr22_with_ERCC92.gtf -o gffcompare stringtie_merged.gtf
cat gffcompare.stats
```

  将合并的注释与已知的注释进行比较后，它是什么样子的?与一般的gtf有什么不同?

  ```
awk '{if($3=="transcript") print}' gffcompare.annotated.gtf | cut -f 1,4,9 | less
  ```

对于de novo模式:

```
ls -1 *Rep*/transcripts.gtf > assembly_GTF_list.txt
cat assembly_GTF_list.txt
stringtie --merge -p 8 -o stringtie_merged.gtf assembly_GTF_list.txt
```

将重新合并的转录本与已知的注释进行比较:

```
gffcompare -r ../chr22_with_ERCC92.gtf -o gffcompare stringtie_merged.gtf
cat gffcompare.stats
```

